blockstore/apps/bundles/store.py

```diff
@@ -208,68 +203,128 @@ class BundleDataStore():
     backing store for running tests.
 
     Advanced: It's possible that one day we'll want to swap out the entire
-    BundleDataStore to adopt a radically different versioning store like git
+    SnapshotRepo to adopt a radically different versioning store like git
     (+LFS). In that case, we'd create a new class that implements the same
     public BundleStore interface. To keep the option open, we'd want to make
     sure that things going in and out of BundleStore don't leak storage
     implementation details.
+
+    # Storage conventions:
+
+    Metadata JSON file describing the contents of a Snapshot and basic metadata:
+        {bundle_uuid}/snapshots/{snapshot_digest.hex()}.json
+
+    Data for each file:
+        {bundle_uuid}/snapshot_data/{file_hash.hexdigest()}
+
+    Notes:
+    * To make S3 partitioning work correctly, we need to lead with the UUID or
+      some other randomly distributed string.
+    * A data file is stored with a hash as a name since it could be referenced
+      by different paths in different versions.
     """
-    def snapshot(self, bundle_uuid, snapshot_digest):
-        """ Return a snapshot. """
-        storage_path = "{}/snapshots/{}.json".format(bundle_uuid, snapshot_digest.hex())
-        with default_storage.open(storage_path) as snapshot_file:
-            snapshot_json = json.load(snapshot_file)
-
-        files = {
-            pathname: FileInfo(pathname, file_info[0], file_info[1], codecs.decode(file_info[2], 'hex'))
-            for pathname, file_info in snapshot_json['files'].items()
-        }
-        hash_digest = codecs.decode(snapshot_json['hash_digest'], 'hex')
-
-        # Python's datetime module is incapable of parsing the ISO 8601
-        # timestamps that it itself produces (it can't do timezones with ":"
-        # in them), so we need to only parse the part before that and then
-        # manually add UTC timezone information into it.
-        created_at = datetime.strptime(
-            snapshot_json['created_at'], '%Y-%m-%dT%H:%M:%S.%f+00:00'
-        )
-        created_at = created_at.astimezone(timezone.utc)
+    def __init__(self, storage=None):
+        self.storage = storage or default_storage
+
+    def get(self, bundle_uuid: UUID, snapshot_digest: bytes) -> Snapshot:
+        """
+        Return a snapshot.
 
-        return BundleSnapshot(
+        TODO: There's no meaningful error handling yet.
+        """
+        storage_path = self._summary_path(bundle_uuid, snapshot_digest)
+        try:
+            with self.storage.open(storage_path) as snapshot_file:
```

Comment:
Storing this summary file on S3 rather than MySQL still seems very odd to me, but let's see how it goes :)